src/components/List/ListItem/_list-item.scss

```diff
@@ -175,7 +175,7 @@
         text-overflow: ellipsis;
         padding-right: $spacing-09;
         &__with-actions {
-          padding-right: 0;
+          padding-right: $spacing-06; // save room for the action button
```

Comment:
You need even more room. The `bx--overflow-menu` is 2rem so you'll need at least `$spacing-07`

Reply:
for some reason that seems to work for me, I think the ellipses padding helps